actionview/lib/action_view/helpers/translation_helper.rb

```diff
@@ -93,7 +93,7 @@ def translate(key, **options)
           break translated unless translated == MISSING_TRANSLATION
 
           if alternatives.present? && !alternatives.first.is_a?(Symbol)
-            break alternatives.first && I18n.translate(**options, default: alternatives)
+            break alternatives.first && I18n.translate(key, **options, default: alternatives)
```

Comment:
`key` was intentionally omitted from this call to avoid (re)looking up a key that is known to be missing.

`I18n.translate` supports omitting the `key` argument: https://github.com/ruby-i18n/i18n/blob/a5a2ba6c6991cf66f733016aea91edd1f0ea1113/lib/i18n.rb#L210 (see `key = nil`).

> I had problems with it and custom I18n debug addon

That sounds like a bug in your addon.  However, if it is just an issue of method arity, we could explicitly pass `nil` here:

```suggestion
            break alternatives.first && I18n.translate(nil, **options, default: alternatives)
```

Reply:
The key argument (as first positional) can not be omitted if other arguments present.

Reply:
I do not understand what you mean.  See:

1. `I18n.translate` calls `translate_key`:
    https://github.com/ruby-i18n/i18n/blob/a5a2ba6c6991cf66f733016aea91edd1f0ea1113/lib/i18n.rb#L222
2. `translate_key` calls `backend.translate`:
    https://github.com/ruby-i18n/i18n/blob/a5a2ba6c6991cf66f733016aea91edd1f0ea1113/lib/i18n.rb#L362
3. `I18n::Backend::Base#translate` supports a `nil` value for `key` even when there are other arguments / options:
    https://github.com/ruby-i18n/i18n/blob/a5a2ba6c6991cf66f733016aea91edd1f0ea1113/lib/i18n/backend/base.rb#L33-L36

Reply:
The key can be nil, but it can not be omitted. There is no reason to call translation without any argument. The default value of key argument in I18n.translation definition just confuses.